ui-tests/test/service-worker.spec.ts

```diff
@@ -104,4 +104,20 @@ test.describe('Service Worker Tests', () => {
     await page.notebook.open(notebook);
     await page.notebook.runCellByCell();
   });
+
+  test('Recreate a file that was deleted from the UI', async ({ page }) => {
+    const notebook = 'file-access-4.ipynb';
+
+    await page.menu.clickMenuItem('Settings>Autosave Documents');
+
+    // Create the file once
+    await page.notebook.open(notebook);
+    await page.notebook.runCellByCell();
+
+    // Delete it from the UI
+    await page.filebrowser.contents.deleteFile('data.json');
+
+    // Recreate it
+    await page.notebook.runCellByCell();
```

Comment:
Does this detect any potential error in the cell output, or would it fail silently?

Reply:
it seems there is no way of doing this with Galata. But now that jupyterlite stops execution on error, I guess `runCellByCell` should be enough to check the notebook runs without error?

Reply:
Since the test is enough to catch the kernel crashing issue we were seeing, I don't think the test is a blocker to merge the PR.

We could track that in an issue in JupyterLab to add galata APIs for actually checking notebook runs.